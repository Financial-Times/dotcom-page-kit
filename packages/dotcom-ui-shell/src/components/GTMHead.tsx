import React from 'react'
import { TFlagsData } from '@financial-times/dotcom-ui-flags/src/types'

// This component is maintained by the ads team
type Props = {
  flags: TFlagsData
  appName?: string
}

// This component is maintained by the ads team
const GTMHead = ({ flags, appName }: Props) => {
  if (!flags.enableGTM) return null

  // 1) If appName is artcile-partner-content use the PC GTM instance
  const PCAppOverride =
    appName === 'article-partner-content' ? 'https://www.googletagmanager.com/gtm.js?id=GTM-12345' : null

  // 2) If appName is not Partner Content, use the standard FT.com GTM isntance and use the first-party GTM domain if flag is set.
  const src =
    PCAppOverride ??
    (flags['ads-first-party-gtm']
      ? 'https://www.ft.com/page-resources/'
      : 'https://www.googletagmanager.com/gtm.js?id=GTM-NWQJW68')

  const tagManager = `(function(w,d,s,l){
    w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
    var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
    j.async=true;
    j.src='${src}'+dl;
    f.parentNode.insertBefore(j,f);
  })(window,document,'script','dataLayer');`

  return <script dangerouslySetInnerHTML={{ __html: tagManager }} />
}

GTMHead.defaultProps = {
  flags: {}
}

export default GTMHead
