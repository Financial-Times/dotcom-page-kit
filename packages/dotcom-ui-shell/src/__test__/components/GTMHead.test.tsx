import React from 'react'
import renderer from 'react-test-renderer'
import GTMHead from '../../components/GTMHead'

describe('dotcom-ui-shell/src/components/GTMHead', () => {
  it('renders the gtm head script when the enableGTM flag is on', () => {
    const props = {
      flags: {
        enableGTM: true
      }
    }

    const tree = renderer.create(<GTMHead {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders null when the enableGTM flag is off', () => {
    const props = {
      flags: {
        enableGTM: false
      }
    }

    const tree = renderer.create(<GTMHead {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
