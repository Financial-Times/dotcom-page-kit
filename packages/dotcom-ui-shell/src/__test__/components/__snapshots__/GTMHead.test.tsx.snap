// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/GTMHead renders null when the enableGTM flag is off 1`] = `null`;

exports[`dotcom-ui-shell/src/components/GTMHead renders the first party gtm script when the ads-first-party-gtm flag is on 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "(function(w,d,s,l){
    w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
    var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
    j.async=true;
    j.src='https://www.ft.com/page-resources/'+dl;
    f.parentNode.insertBefore(j,f);
  })(window,document,'script','dataLayer');",
    }
  }
/>
`;

exports[`dotcom-ui-shell/src/components/GTMHead renders the gtm head script when the enableGTM flag is on 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "(function(w,d,s,l){
    w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
    var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
    j.async=true;
    j.src='https://www.googletagmanager.com/gtm.js?id=GTM-NWQJW68'+dl;
    f.parentNode.insertBefore(j,f);
  })(window,document,'script','dataLayer');",
    }
  }
/>
`;
